File: 3731-find-missing-elements/solution.cpp
#include <vector>
#include <algorithm>
#include <unordered_set>

class Solution {
public:
    std::vector<int> findMissingIntegers(std::vector<int>& nums) {
        // Find the minimum and maximum values in the array
        int min_val = *std::min_element(nums.begin(), nums.end());
        int max_val = *std::max_element(nums.begin(), nums.end());
        
        // Insert all numbers into a hash set for O(1) lookup
        std::unordered_set<int> present(nums.begin(), nums.end());
        
        std::vector<int> missing;
        
        // Iterate through the entire range [min_val, max_val]
        for (int i = min_val; i <= max_val; ++i) {
            // If the integer is not found in the set, it is missing
            if (present.find(i) == present.end()) {
                missing.push_back(i);
            }
        }
        
        return missing;
    }
};

File: 3731-find-missing-elements/solution.py
from typing import List

class Solution:
    def findMissingIntegers(self, nums: List[int]) -> List[int]:
        # Determine the boundaries of the original contiguous range
        min_val, max_val = min(nums), max(nums)
        
        # Store existing numbers in a set for O(1) average lookup time
        present = set(nums)
        
        # Collect and return all numbers in [min_val, max_val] not present in nums
        return [x for x in range(min_val, max_val + 1) if x not in present]